internal/docker-client/worker-event-loop.go

```diff
@@ -0,0 +1,16 @@
+package docker_client
+
+import (
+	"context"
+	"github.com/Capstone-auto-grader/grader-api-v2/internal/grader-task"
+	sync_map "github.com/Capstone-auto-grader/grader-api-v2/internal/sync-map"
+)
+
+func WorkerEventLoop(queue <-chan string, client *DockerClient, tasks *sync_map.SyncMap) {
+	for {
+		taskId := <-queue
+		_ = tasks.UpdateStatus(taskId, grader_task.StatusStarted, true)
+		_ = client.StartTask(context.Background(), taskId)
+		_,_ = client.TaskOutput(context.Background(), taskId)
```

Comment:
I think we can make another channel here for propagating errors out.

Reply:
I was thinking a channel for all responses and errors, so another worker can deal with them